at_client/src/main/java/org/atsign/client/util/KeysUtil.java

```diff
@@ -18,68 +18,104 @@
 public class KeysUtil {
     private static final ObjectMapper mapper = new ObjectMapper();
 
-    private static final String rootFolder = System.getProperty("user.dir") + "/keys/";
+    private static final String rootFolder = System.getProperty("user.home") + "/.atsign/keys/";
+    public static final String keysFileSuffix = "_key.atKeys";
 
-    public  static final String pkamPublicKeyName = "aesPkamPublicKey";
-    public  static final String pkamPrivateKeyName = "aesPkamPrivateKey";
-    public  static final String encryptionPublicKeyName = "aesEncryptPublicKey";
-    public  static final String encryptionPrivateKeyName = "aesEncryptPrivateKey";
-    public  static final String selfEncryptionKeyName = "selfEncryptionKey";
+    public static final String pkamPublicKeyName = "aesPkamPublicKey";
+    public static final String pkamPrivateKeyName = "aesPkamPrivateKey";
+    public static final String encryptionPublicKeyName = "aesEncryptPublicKey";
+    public static final String encryptionPrivateKeyName = "aesEncryptPrivateKey";
+    public static final String selfEncryptionKeyName = "selfEncryptionKey";
 
     public static void saveKeys(AtSign atSign, Map<String, String> keys) throws Exception {
-        _makeRootFolder();
-        File file = getKeysFile(atSign);
-        System.out.println("Saving keys to " + file.getAbsolutePath());
-        if (file.exists()) {
-            //noinspection ResultOfMethodCallIgnored
+        File file = getKeysFile(atSign, rootFolder);
+        // First check for keys file at ~/.atsign/keys/$atSign_key.atKeys
+        // then check at ./keys/$atSign_key.atKeys
+        if (!file.exists()) {
+            String keysFileName = System.getProperty("user.dir") + "/keys/" + atSign + keysFileSuffix;
+            file = getKeysFile(atSign, keysFileName);
+            if (!file.exists()) {
+                // If atKeys file does not exist,
+                // create dir in ~/.atsign/keys/
+                // to store key files
+                _makeRootFolder();
```

Comment:
Currently, _makeRootFolder() will throws an exception. There's no need to throw an exception here, because we know what to do. We've checked if the file already exists somewhere and it doesn't - so now, we need to create the root folder (if it doesn't exist already), then create the keys file within the root folder using getKeysFile(atSign, rootFolder)

Reply:
Just creating a list of what you would like to be updated:
- Remove exception thrown in _makeRootFolder()
- Add func getKeysFile(atSign, rootFolder) within _makeRootFolder()

Does that look right to you? @gkc

Reply:
First one yes, second one no